extension/chrome/settings/index.ts

```diff
@@ -151,7 +151,12 @@ Catch.try(async () => {
       }
     }
 
-    await renderBlogPostList();
+    Api.retreiveBlogPosts().then((res: { url: string, date: string, title: string }[]) => {
```

Comment:
Much better! Now, don't clarify the type of `res` here when receiving it. That should be clarified in the api file with `retreiveBlogPosts = async (): Promise<R.FcBlogPosts> => {`

Then here you only do `Api.retreiveBlogPosts().then(res => {`

This is so that if we later want to change the structure of the api response, you don't have to go hunting for all usages of this function where you would have to change the blog post structure. There will only be one authoritative place defining the response type, in the api file.

Reply:
Also I recommend to use more descriptive `Api.retreiveBlogPosts().then(blogs => {`